apps/client/src/features/app-settings/panel/project-panel/ProjectListItem.tsx

```diff
@@ -0,0 +1,165 @@
+import { Menu, MenuButton, IconButton, MenuList, MenuItem } from '@chakra-ui/react';
+import { IoEllipsisHorizontal } from '@react-icons/all-files/io5/IoEllipsisHorizontal';
+import { useEffect, useMemo, useState } from 'react';
+import { AxiosError } from 'axios';
+
+import style from './ProjectPanel.module.scss';
+import { renameProject, loadProject, duplicateProject } from '../../../../common/api/ontimeApi';
+import { ontimeQueryClient } from '../../../../common/queryClient';
+import { PROJECT_LIST } from '../../../../common/api/apiConstants';
+import { EditMode } from './ProjectList';
+import RenameProjectForm, { RenameProjectFormValues } from './RenameProjectForm';
+import DuplicateProjectForm, { DuplicateProjectFormValues } from './DuplicateProjectForm';
+
+interface ProjectListItemProps {
+  current?: boolean;
+  filename: string;
+  createdAt: string;
+  updatedAt: string;
+  onToggleEditMode?: (editMode: EditMode, filename: string | null) => void;
+  onSubmit?: () => void;
+  editingFilename: string | null;
+  editingMode: EditMode | null;
+}
+
+export default function ProjectListItem({
+  current,
+  createdAt,
+  editingFilename,
+  editingMode,
+  filename,
+  onSubmit,
+  onToggleEditMode,
+  updatedAt,
+}: ProjectListItemProps) {
+  const [submitError, setSubmitError] = useState<string | null>(null);
+
+  const handleRefetch = async () => {
+    await ontimeQueryClient.invalidateQueries({ queryKey: PROJECT_LIST });
+  };
+
```

Comment:
nice! 
We also need to invalidate the remaining keys once a project is loaded. I think there is a `invalidateAllCaches` function

Reply:
question

if we do `invalidateAllCaches` do we still need to run 

`await ontimeQueryClient.invalidateQueries({ queryKey: PROJECT_LIST });`

on project load? I don't think so right?

Reply:
Since `PROJECT_LIST` is new it might not be in `invalidateAllCaches`

I guess if we add it there we are all good.

The gist of it is to tell react with "hey! The whole dataset just changed, if you need stuff fetch it again". There might be a single function to invalidate the entire `queryClient`

Reply:
makes sense

I've seen this on react-query docs:

```
// Invalidate every query in the cache
queryClient.invalidateQueries()
```

https://tanstack.com/query/latest/docs/react/guides/query-invalidation

should we just invalidate all with `queryClient.invalidateQueries()` or is there a reason we are passing the keys individually to our invalidate function?

Reply:
I think we should use the function stated in the documentation.

I think the reason might well have been a "did-not-read-the-docs" situation 😅